extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/multipart/MultipartResponseTest.java

```diff
@@ -0,0 +1,280 @@
+        assertThat(data.file).exists();
+        assertThat(data.name).isEqualTo("foo");
+        assertThat(data.panda.weight).isEqualTo("huge");
+        assertThat(data.panda.height).isEqualTo("medium");
+        assertThat(data.panda.mood).isEqualTo("happy");
+    }
+
+    void verifyWooHooFile(File file) {
+        int position = 0;
+        try (FileReader reader = new FileReader(file)) {
+            int read;
+            while ((read = reader.read()) > 0) {
+                assertThat((char) read).isEqualTo(WOO_HOO_WOO_HOO_HOO.charAt(position % WOO_HOO_WOO_HOO_HOO.length()));
+                position++;
+            }
+            assertThat(position).isEqualTo(WOO_HOO_WOO_HOO_HOO.length() * 10000);
+        } catch (IOException e) {
+            fail("failed to read provided file", e);
+        }
+    }
+
+    @Path("/give-me-file")
+    public interface Client {
+        @GET
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        MultipartData getFile();
+
+        @GET
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        @Path("/empty")
+        MultipartData getFileEmpty();
+
+        @GET
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        MultipartDataWithSetters getFileWithSetters();
+
+        @GET
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        @Path("/error")
+        MultipartData error();
+    }
+
+    @Path("/give-me-file")
+    public static class Resource {
+
+        @GET
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        public MultipartData getFile() throws IOException {
+            File file = File.createTempFile("toDownload", ".txt");
+            file.deleteOnExit();
+            // let's write Woo hoo, woo hoo hoo 10k times
+            try (FileOutputStream out = new FileOutputStream(file)) {
+                for (int i = 0; i < 10000; i++) {
+                    out.write(WOO_HOO_WOO_HOO_HOO.getBytes(StandardCharsets.UTF_8));
+                }
+            }
+            return new MultipartData("foo", file, new Panda("huge", "medium", "happy"),
+                    1984, new int[] { 2008, 2011, 2014 });
+        }
+
+        @GET
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        @Path("/empty")
+        public MultipartData getEmptyData() {
+            return new MultipartData(null, null, null, 0, null);
+        }
+
+        @GET
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        @Path("/error")
+        public MultipartData throwError() {
+            throw new RuntimeException("forced error");
+        }
+    }
+
+    public static class MultipartData {
+
+        @RestForm
+        @PartType(MediaType.TEXT_PLAIN)
+        public String name;
+
+        @RestForm
+        @PartType(MediaType.APPLICATION_OCTET_STREAM)
+        public File file;
+
+        @RestForm
+        @PartType(MediaType.APPLICATION_JSON)
+        public Panda panda;
+
+        @RestForm
+        @PartType(MediaType.TEXT_PLAIN)
+        public Integer number;
+
+        @RestForm
```

Comment:
I would change only this field to use `FormParam("numberz")` just for coverage purposes (it should work in the same way).

Reply:
good catch, it actually doesn't work, fixing